pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -317,11 +324,19 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 			}
 			log.Info("machine has current version", "name", machine.GetName(),
 				"version", node.Annotations[nodeconfig.VersionAnnotation])
-			// version annotation exists with a valid value, node is fully configured, do nothing.
+			// version annotation exists with a valid value, node is fully configured.
+			// configure prometheus if it is not configured
+			if err := pc.Configure(); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "unable to configure Prometheus")
+			}
 			return reconcile.Result{}, nil
 		}
 	} else if *machine.Status.Phase != provisionedPhase {
 		log.V(1).Info("machine not provisioned", "phase", *machine.Status.Phase)
+		// configure prometheus if it is not configured
+		if err := pc.Configure(); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "unable to configure Prometheus")
```

Comment:
Why are we doing this for Machines that are not in the `provisionedPhase`? This implies the comment should explain why it is needed.